src/jit/lower.cpp

```diff
@@ -240,6 +240,22 @@ GenTree* Lowering::LowerNode(GenTree* node)
 
         case GT_STORE_BLK:
         case GT_STORE_OBJ:
+            if (node->AsBlk()->gtBlkSize == 0)
+            {
+                // Storing a block of zero size is a no-op.
+                GenTree*           nextNode = node->gtNext;
+                bool               isClosed;
+                LIR::ReadOnlyRange blockStoreRange = BlockRange().GetTreeRange(node, &isClosed);
+                if (isClosed)
+                {
+                    JITDUMP("Removing zero-length block store:\n");
+                    DISPTREE(node);
+                    BlockRange().Remove(std::move(blockStoreRange));
```

Comment:
As I see CodeGen is prepared to see zero size store block (when `isClosed == false` or  `GT_STORE_BLK` appeared later somehow), but I do not understand how can we get `isClosed == false`, is it possible?

Reply:
I believe that `isClosed` could be false if part of the expression had a comma expression that was linearized in-between the evaluation of the operands and the block store. It seems, though, that this needs further work anyway to ensure that we don't remove side-effects. That, and the desire to avoid introducing risk into 3.0, is why I removed this part and filed #24953.